game/theater/theatergroundobject.py

```diff
@@ -90,11 +92,24 @@ def dead_units(self) -> List[Unit]:
             )
         )
 
+    # This function returns the name of the whole group
+    # will be used for the mission generator as dcs unit name
     @property
     def group_name(self) -> str:
         """The name of the unit group."""
         return f"{self.category}|{self.group_id}"
 
+    # if the TGO has more than 1 group this function will be used
+    # it returns the unique group name of the specific group
+    # We currently have no tgo / global naming convention for the groups
+    # the group names will be set by the generator
+    # example: add_auxiliary_group for huge sam groups like patriots
+    def single_group_name(self, group_index: int) -> str:
+        try:
+            return self.groups[group_index].name
+        except IndexError:  # handling for possible backwards incompatibility
```

Comment:
How is this save compat? `groups` was already a member. If the size of the group changed across saves something went wrong and it's just a bug.